packages/connect-form/src/utils/tls-options.ts

```diff
@@ -0,0 +1,41 @@
+import ConnectionStringUrl from 'mongodb-connection-string-url';
+import { ConnectionOptions } from 'mongodb-data-service';
+
+import { TLS_OPTIONS } from '../constants/ssl-tls-options';
+import { ConnectionFormError } from '../utils/connect-form-errors';
+
+export function handleUpdateTlsOption({
+  tlsOption,
+  connectionStringUrl,
+  connectionOptions,
+}: {
+  tlsOption: TLS_OPTIONS;
+  connectionStringUrl: ConnectionStringUrl;
+  connectionOptions: ConnectionOptions;
+}): {
+  connectionStringUrl: ConnectionStringUrl;
+  connectionOptions: ConnectionOptions;
+  errors: ConnectionFormError[];
+} {
+  const updatedConnectionString = connectionStringUrl.clone();
+
+  if (tlsOption === TLS_OPTIONS.ON) {
+    updatedConnectionString.searchParams.delete('ssl');
+    updatedConnectionString.searchParams.set('tls', 'true');
+  } else if (tlsOption === TLS_OPTIONS.OFF) {
+    updatedConnectionString.searchParams.delete('ssl');
+    updatedConnectionString.searchParams.set('tls', 'false');
+  } else if (tlsOption === TLS_OPTIONS.DEFAULT) {
+    updatedConnectionString.searchParams.delete('ssl');
+    updatedConnectionString.searchParams.delete('tls');
```

Comment:
fyi, if https://github.com/mongodb-js/compass/pull/2651 is merged first it might be nice to update the core here to have similar changes :)